replace.go

```diff
@@ -149,10 +149,6 @@ func (r *Runner) ReplaceWithRendered(name, chartName, chartPath string, o Replac
 		return nil, fmt.Errorf("invalid state: no files rendered")
 	}
 
-	if err := os.RemoveAll(helmOutputDir); err != nil {
-		return nil, fmt.Errorf("cleaning up unnecessary files after replace: %v", err)
-	}
```

Comment:
We still need this block for cleanup when your chart is located at a place other than `./`.

https://github.com/variantdev/chartify/pull/13#issuecomment-1013647811

See L139-L141- what it does there is to rename anything under `$TEMP_DIR/helmx1.rendered/$SUB_DIR` to `$TEMP_DIR/$SUB_DIR`, so that we can effectively "replace" the chart contained in `$TEMP_DIR` with the rendering result of the chart. We only use `$TEMP_DIR/helmx1.rendered` for a temporary location of the intermediate chart output, which can be removed after L147.

But `chartName` of `.` breaks that's behavior. The important part is L70-L76:

```
	helmOutputDir := filepath.Join(chartPath, "helmx.1.rendered")
	if err := os.Mkdir(helmOutputDir, 0755); err != nil {
		return nil, err
	}
	// This directory contains templates/ and charts/SUBCHART/templates
	chartOutputDir := filepath.Join(helmOutputDir, chartName)
```

So this function works correct only when `helmOutputDir != chartOutputDir`, because we move everything under `helmOutputDir` to `chartOutputDir` and remove `helmOutputDir`.

But `chartName` of `.` makes it `helmOutputDir == chartOutputDir`, which ends up removing the whole output dir which must exist for later process.

Reply:
I believe the correct fix would be to modify L76 instead.

More concretely, in the below code:

```
chartOutputDir := filepath.Join(helmOutputDir, chartName)
```

`chartOuputDir` shouldn't result in `helmOutputDir` if `chartName` is `.`.
Probably we'd better get the chartName by parsing `Chart.yaml` instead? Probably that's what `helm template --output-dir` appends to the output dir path.

Reply:
Hi @mumoshu 
Thats an interesting edge case but honestly i was not looking for that. I am facing same symptoms on windows when trying to use transformers and by debugging and using simple test case on linux i thought i found the rootcause of the issue. I would have to continue to debug also on windows to understand what could be the problem there. Would you mind to keep this topic open?

thx 
tom

Reply:
> Probably we'd better get the chartName by parsing `Chart.yaml` instead? Probably that's what `helm template --output-dir` appends to the output dir path.

If chartName is not from Chart.yaml, I think that explains my problem in https://github.com/variantdev/chartify/pull/13#issuecomment-1039090849

Reply:
See also https://github.com/helm/helm-www/blob/5ce1672465102890a16a3f9ef6549444084f6df1/content/en/docs/topics/charts.md#the-chart-file-structure

Reply:
Thanks for sharing.
An another important point for this is that there can be cases that `Chart.yaml` is missing, for example when you're turning raw k8s manifests or kustomize config into a chart using chartify.
Perhaps a better approach would be to use use the last non empty basename of the path(so if chart is `foo/bar/../.` it should use `bar` as the chart name)